homeassistant/components/airzone/climate.py

```diff
@@ -156,14 +157,13 @@ def _async_update_attrs(self) -> None:
         """Update climate attributes."""
         self._attr_current_temperature = self.get_airzone_value(AZD_TEMP)
         self._attr_current_humidity = self.get_airzone_value(AZD_HUMIDITY)
+        self._attr_hvac_action = HVAC_ACTION_LIB_TO_HASS[
+            self.get_airzone_value(AZD_ACTION)
+        ]
         if self.get_airzone_value(AZD_ON):
-            mode = self.get_airzone_value(AZD_MODE)
-            self._attr_hvac_mode = HVAC_MODE_LIB_TO_HASS[mode]
-            if self.get_airzone_value(AZD_DEMAND):
-                self._attr_hvac_action = HVAC_ACTION_LIB_TO_HASS[mode]
-            else:
-                self._attr_hvac_action = HVACAction.IDLE
+            self._attr_hvac_mode = HVAC_MODE_LIB_TO_HASS[
+                self.get_airzone_value(AZD_MODE)
+            ]
         else:
-            self._attr_hvac_action = HVACAction.OFF
```

Comment:
Should this still be set?

It looks like the climate platform will still set this value even if hvac mode is off.

https://github.com/home-assistant/core/blob/809f394ff5425ca93876fe9bea75614a82b50044/homeassistant/components/climate/__init__.py#L335

Reply:
This is now handled in the library, so it will set it to off if get_on() is False:
https://github.com/Noltari/aioairzone/blob/master/aioairzone/device.py#L666